### PR TITLE
Add flag for enabling/disabling mlockall

### PR DESCRIPTION
**Pull request by csstaub**
r @alokmenghrajani 
Add flag for enabling/disabling mlockall

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -34,20 +34,21 @@ import (
...
+	caFile        = app.Flag("ca", "PEM-encoded CA certificates file").Required().String()
+	asuser        = app.Flag("asuser", "Default user to own files").Default("keywhiz").String()
+	asgroup       = app.Flag("group", "Default group to own files").Default("group").String()
+	ping          = app.Flag("ping", "Enable startup ping to server").Default("false").Bool()
+	debug         = app.Flag("debug", "Enable debugging output").Default("false").Bool()
+	timeout       = app.Flag("timeout", "Timeout for communication with server").Default("20s").Duration()
+	metricsURL    = app.Flag("metrics-url", "Collect metrics and POST them periodically to the given URL (via HTTP/JSON).").String()
+	metricsPrefix = app.Flag("metrics-prefix", "Override the default metrics prefix used for reporting metrics.").String()
+	syslog        = app.Flag("syslog", "Send logs to syslog instead of stderr.").Default("false").Bool()
+	enableMlock   = app.Flag("mlock", "Call mlockall on process memory.").Default("true").Bool()
```
Comment (mcpherrinm) [inline main.go]: Can we default to false?  All our other bool flags do

**Comment by mcpherrinm** [comment]
lgtm, but would prefer enableMlock to default off: I'm not convinced it adds a ton of value in our swap-free world, and apps will swap to disk with secrets anyways since they don't mlock probably.

File: main.go (hunk 1 above)
Comment (csstaub) [inline main.go]: I want to preserve the current default behavior. I inverted the flag logic though.

**Comment by alokmenghrajani** [comment]
🚢

**Comment by mcpherrinm** [comment]
LGTM now

**Pull request merged**